src/core/turn-order.js

```diff
@@ -51,79 +51,68 @@ export function SetActivePlayers(ctx, playerID, arg) {
     _prevActivePlayers = [];
   }
 
-  let activePlayers = {};
+  let localCtx = {
```

Comment:
I find it a bit confusing to use `ctx` to represent `{ activePlayers, _activePlayersMoveLimit }`.  I can see the desire to club them together for the sake of `ApplyActivePlayerArgument`.

Shall we just pass them individually to `ApplyActivePlayerArgument` like this:

```js
ApplyActivePlayerArgument(activePlayers, _activePlayersMoveLimit, playerID, value);
```

and have the function mutate them directly?  We're passing in new objects anyway, so we won't be violating any immutability principle here.

Reply:
Sure. I tried sticking to the other related signatures — `(ctx, playerID, arg)` — but I was a little unhappy with how this felt too.

That said, given we’re returning `{ activePlayers, _activePlayersMoveLimit }`, I think it’s nice to pass that in in that form already. Also because otherwise in `setActivePlayers` we have to do this each time:

```js
({ activePlayers, _activePlayersMoveLimit } = ApplyActivePlayerArgument(
  activePlayers, _activePlayersMoveLimit, playerID, arg
));
```

Reply:
Do we need to have `ApplyActivePlayerArgument` return anything at all, though?  Can it not just mutate `activePlayers` and `_activePlayersMoveLimit` directly?

Reply:
Oh, I see, you’re suggesting mutating directly, so we’d use it simply like this:

```js
ApplyActivePlayerArgument(activePlayers, _activePlayersMoveLimit, playerID, arg);
```

OK 😀

Reply:
Done in  07ca81c